### PR TITLE
[2.0] fix CI and updated 2.0 whats new

### LINKED ISSUE #4911: Update the What's New for 2.0 to be consistent with earlier releases
All of our "What's New" pages – except for the 2.0 one – have an involved Overview section that outlines the big changes and contains a "by the numbers" section.  The What's New for 2.0 should be updated to match the style of the others.

The "by the numbers" for 2.0 can be found in the current `RELEASE.rst`
(https://github.com/sunpy/sunpy/blame/a3cb0cea07d0972486f20dd78e2f528f1fe27c04/RELEASE.rst#L22):
> This release of SunPy contains 1044 commits in 290 merged pull requests closing 144 issues from 33 people, 16 of which are first-time contributors to SunPy.

### PR DESCRIPTION
**Pull request by nabobalis**
Fixes #4911

### CODE REVIEW

File: docs/whatsnew/2.0.rst [hunk 1]
```diff
@@ -10,7 +10,29 @@ Overview
...
+* :ref:`whatsnew-2.0-python`
+* :ref:`whatsnew-2.0-search`
+* :ref:`whatsnew-2.0-aiaprep`
+* :ref:`whatsnew-2.0-pixelindex`
+* :ref:`whatsnew-2.0-standards`
+* :ref:`whatsnew-2.0-overview`
+* :ref:`whatsnew-2.0-coordframe`
+* :ref:`whatsnew-2.0-carrington`
+* :ref:`whatsnew-2.0-proxy`
+* :ref:`whatsnew-2.0-citation`
```
Comment (dstansby) [inline docs/whatsnew/2.0.rst:24]: Why not just add a local table of contents?

**Review by dstansby — APPROVED** [review]
Seems fine - is there a reason we haven't got rid of the devedeps build on 2.0 yet? (if it's just no-ones bothered to yet, I can put in a PR)

File: docs/whatsnew/2.0.rst (hunk 1 above)
Comment (nabobalis) [reply docs/whatsnew/2.0.rst:24]: Good question.

**Comment by nabobalis** [comment]
> Seems fine - is there a reason we haven't got rid of the devedeps build on 2.0 yet? (if it's just no-ones bothered to yet, I can put in a PR)

Devdeps are still important to check. Its broken to a test returning the wrong type but I am unsure why its only for 2.0 and not 2.1

**Comment by nabobalis** [comment]
Online fails are real due to my changes.

**Comment by ayshih** [comment]
> Devdeps are still important to check. Its broken to a test returning the wrong type but I am unsure why its only for 2.0 and not 2.1

2.1 falls too (see daily cron)

File: sunpy/net/dataretriever/sources/noaa.py [hunk 2]
```diff
@@ -31,13 +31,11 @@ class NOAAIndicesClient(GenericClient):
...
     <BLANKLINE>
-    1 Results from the NOAAIndicesClient:
-         Start Time           End Time      Source  Instrument  Wavelength
-    ------------------- ------------------- ------ ------------ ----------
-    2016-01-01 00:00:00 2016-01-02 00:00:00   sdic noaa-indices        nan
+    0 Results from the VSOClient:
+    Start Time End Time Source Instrument Type
+    ---------- -------- ------ ---------- ----
     <BLANKLINE>
     <BLANKLINE>
```
Comment (nabobalis) [inline sunpy/net/dataretriever/sources/noaa.py:38]: This is odd.

Comment (nabobalis) [reply sunpy/net/dataretriever/sources/noaa.py]: Fixed.

File: sunpy/net/dataretriever/sources/noaa.py [hunk 3]
```diff
@@ -72,7 +71,7 @@ def _can_handle_query(cls, *query):
         chkattr = ['Time', 'Instrument']
         chklist = [x.__class__.__name__ in chkattr for x in query]
         for x in query:
-            if x.__class__.__name__ == 'Instrument' and x.value == 'noaa-indices':
+            if x.__class__.__name__ == 'Instrument' and x.value.lower() == 'noaa-indices':
```
Comment (nabobalis) [inline sunpy/net/dataretriever/sources/noaa.py:74]: WOOPS

**Comment by nabobalis** [comment]
Bunch of things changed:

1. Added py39 wheels and test suite
2. Remembered we don't run the doctests properly in the online suite so I fixed that and any broken tests.
3. Found a bug in the NOAA-INDICES client.

File: azure-pipelines.yml [hunk 4]
```diff
@@ -53,15 +53,14 @@ jobs:
         - python-devel
 
     envs:
-      - macos: py36
-        name: py36_test
-        posargs: -n=1
+      - macos: py38
+        name: py38_test
```
Comment (dstansby) [inline azure-pipelines.yml:57]: Why has this changed?

File: azure-pipelines.yml [hunk 5]
```diff
@@ -53,15 +53,14 @@ jobs:
...
 
       - windows: py37
         name: py37_test
 
-      - linux: py38-online
-        name: py38_test_online
+      - linux: py39-online
+        name: py39_test_online
```
Comment (dstansby) [inline azure-pipelines.yml:63]: Adding python 3.9 support should get a changelog entry.

File: docs/whatsnew/2.0.rst (hunk 1 above)
Comment (dstansby) [reply docs/whatsnew/2.0.rst:24]: Is the plan to leave it like this, or add a contents?

File: setup.cfg [hunk 6]
```diff
@@ -100,6 +100,58 @@ remote_data_strict = True
...
+    ignore:ERFA function "dtf2d"*
+    # Currently helio makes unverified requests - this filter should be removed when
+    # https://github.com/sunpy/sunpy/issues/4401 is fixed
+    ignore:Unverified HTTPS request
+    # https://github.com/astropy/astropy/issues/11272
+    ignore:elementwise == comparison failed and returning scalar instead:FutureWarning:astropy.table
+    # This isn't really an issue it's an annoyance, it's triggered by printing
+    # an astropy table inside a test, which is generally only done when
+    # debugging.
+    ignore:the imp module is deprecated in favour of importlib:DeprecationWarning:ipykernel.iostream
```
Comment (dstansby) [inline setup.cfg:154]: Why add these lines? We're still ignoring warnings on the 2.0.x branch.

**Review by dstansby — CHANGES_REQUESTED** [review]
As well as my individual comments, the changelog entries from everything that has been backported also need to be added to this PR.

File: azure-pipelines.yml (hunk 4 above)
Comment (nabobalis) [reply azure-pipelines.yml:57]: Don't need it anymore.

File: azure-pipelines.yml (hunk 5 above)
Comment (nabobalis) [reply azure-pipelines.yml:63]: Sure.

File: setup.cfg (hunk 6 above)
Comment (nabobalis) [reply setup.cfg:154]: It came with a backport and I cba to remove it.

File: docs/whatsnew/2.0.rst (hunk 1 above)
Comment (nabobalis) [reply docs/whatsnew/2.0.rst:24]: Yes.

**Comment by nabobalis** [comment]
> As well as my individual comments, the changelog entries from everything that has been backported also need to be added to this PR.

If a PR that was backported had a changelog, it should have come with the backport.

File: setup.cfg (hunk 6 above)
Comment (nabobalis) [inline setup.cfg:154]:
```suggestion
```

Comment (nabobalis) [reply setup.cfg]: Removed.

File: azure-pipelines.yml (hunk 4 above)
Comment (nabobalis) [reply azure-pipelines.yml:57]: Oh you mean the Python version? We had too many 36 builds, I thought I would bump macos to a more modern version.

**Comment by dstansby** [comment]
> > As well as my individual comments, the changelog entries from everything that has been backported also need to be added to this PR.
> 
> If a PR that was backported had a changelog, it should have come with the backport.

In that case could you add a list of all the PRs that have been backported to this PR description, just so we have a paper trail of which other PRs are being backported?

**Comment by nabobalis** [comment]
> > > As well as my individual comments, the changelog entries from everything that has been backported also need to be added to this PR.
> > 
> > 
> > If a PR that was backported had a changelog, it should have come with the backport.
> 
> In that case could you add a list of all the PRs that have been backported to this PR description, just so we have a paper trail of which other PRs are being backported?

Its in the commit names?

**Comment by dstansby** [comment]
Ah I didn't realise, 👍

**Review by dstansby — APPROVED** [review]

**Pull request merged**